### PR TITLE
Tests for input validation for keras models

### PR DESCRIPTION
**Pull request by GZHoffie**
This PR tests #3927.
Current issues:

1. With `keras==2.3`, I encountered the following error while running test,
   
   ```python
   def setup_method(self, method):
           """
           Setup any state tied to the execution of the given method in a class.
           It is invoked for every test method of a class.
           """
   >       K.set_image_dim_ordering("th")
   E       AttributeError: module 'keras.backend' has no attribute 'set_image_dim_ordering'

   ../pipeline/utils/test_utils.py:40: AttributeError
   ```

   This can be solved by downgrading `keras` to `1.1.0`. For a newer version of `keras`, this should be replaced with `K.common.set_image_dim_ordering('th')`.

2. Several `AssertionError`s occurred as I run the unit tests, as shown in my last commit. Need further inspection on this.
3. Further tests regarding other cases of invalid data will be pushed soon.

### CODE REVIEW

File: pyzoo/zoo/tfpark/tf_dataset.py [hunk 1]
```diff
@@ -1273,6 +1273,34 @@ def __init__(self, df, feature_cols, labels_cols=None, batch_size=-1,
...
+        assert len(structure) == len(names), err_msg
+
+    assert len(names) == 1, f"data does not match {data_type}, " \
+                                  f"data {structure}, {data_type} {names}"
+
+
+def check_data_compatible(dataset, model, mode):
+    input_names = model.input_names
+    output_names = model.output_names
+    err_msg = f"each element is dataset should a tuple for {mode}, " \
```
Comment (GZHoffie) [inline pyzoo/zoo/tfpark/tf_dataset.py:1294]: Should this line be "Each element in dataset should be a tuple for {mode}, "?

Comment (yangw1234) [reply pyzoo/zoo/tfpark/tf_dataset.py:1294]: yes

**Comment by yangw1234** [comment]
Generally LGTM.

Needs to fix coding style http://10.239.176.111:18888/job/ZOO-PR-Python-Spark-2.2-py36/5619/console

**Comment by GZHoffie** [comment]
It seems that the unit test in `test_text_models.py` is failing in jenkins http://10.239.176.111:18888/job/ZOO-PR-Python-Spark-2.1-py36/5629//console. I wasn't able to capture that because I had trouble installing necessary packages like `nlp_architect`.
```python
______________________ TestTextModels.test_intent_entity _______________________

self = <test.zoo.tfpark.test_text_models.TestTextModels testMethod=test_intent_entity>

    def test_intent_entity(self):
        model = IntentEntity(num_intents=8, num_entities=5, word_length=10,
                             word_vocab_size=200, char_vocab_size=50)
        input_data = [np.random.randint(200, size=(8, 30)), np.random.randint(50, size=(8, 30, 10))]
>       output = model.predict(input_data, distributed=True)

../test/zoo/tfpark/test_text_models.py:35: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
../zoo/tfpark/model.py:278: in predict
    results = self._predict_distributed(dataset).collect()
../zoo/tfpark/model.py:292: in _predict_distributed
    predictor = TFPredictor.from_keras(self.model, x)
../zoo/tfpark/tf_predictor.py:74: in from_keras
    check_data_compatible(dataset, keras_model, mode="inference")
../zoo/tfpark/tf_dataset.py:1304: in check_data_compatible
    _check_compatible(input_names, feature, data_type="model_input")
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

names = ['words_input', 'word_chars_input']
structure = TensorMeta(dtype: int64, name: words_input, shape: (30,))
data_type = 'model_input'

    def _check_compatible(names, structure, data_type="model_input"):
        if isinstance(structure, dict):
            err_msg = f"all {data_type} names should exist in data, " \
                      f"got {data_type} {names}, data {structure}"
            assert all([name in structure for name in names]), err_msg
        elif isinstance(structure, list) or isinstance(structure, tuple):
            err_msg = f"{data_type} number does not match data number, " \
                      f"got {data_type} {names}, data {structure}"
            assert len(structure) == len(names), err_msg
        else:
>           assert len(names) == 1, f"data does not match {data_type}, " \
                                    f"data {structure}, {data_type} {names}"
E           AssertionError: data does not match model_input, data TensorMeta(dtype: int64, name: words_input, shape: (30,)), model_input ['words_input', 'word_chars_input']

../zoo/tfpark/tf_dataset.py:1292: AssertionError
```

**Comment by yangw1234** [comment]
I think it is caused by your change here https://github.com/intel-analytics/analytics-zoo/pull/3940/files#diff-b45bdcc1c799a3fcb5bfb6d48fc76ea2052eb5d203756a23c33e7c0b98eb1c84R1178

**Pull request closed without merging**